src/map/itemdb.cpp

```diff
@@ -2997,11 +2996,46 @@ uint8 ItemGroupDatabase::pc_get_itemgroup(uint16 group_id, bool identify, map_se
 	}
 	if (group->random.empty())
 		return 0;
-	
-	// Get all the 'must' item(s) (subgroup 0)
+
+	size_t count = 0;
+	// Count all the 'must' item(s) (subgroup 0)
```

Comment:
```suggestion
	size_t count = 0;
	// Count all the 'must' item(s) (subgroup 0)
```
I would move that above the "if" that you added below, where you really start counting.

Reply:
I can't move it there, because you can have item boxes without must items, like Old Purple Box, Old Card Album...